python/lsst/ts/atpneumaticssimulator/command_issued.py

```diff
@@ -0,0 +1,89 @@
+# This file is part of ts_atpneumaticssimulator.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CommandIssued"]
+
+import asyncio
+
+from lsst.ts import salobj, utils
+
+
+class CommandIssued:
+    """asyncio future to track the progress of an issued low-level controller
+    command.
+
+    Parameters
+    ----------
+    timestamp : float
+        The TAI unix timestamp [sec] at which the command was issued.
+    name : str
+        The name of the command.
+
+    Attributes
+    ----------
+    ack_timestamp : float
+        The TAI unix timestamp [sec] at which an ACK or NOACK was received.
+    done_timestamp : float
+        The TAI unix timestamp [sec] at which a SUCCESS or FAIL was received.
+    done : `asyncio.Future`
+        Future which ends as follows:
+
+        * result = None when the command has ended successfully.
+        * salobj.ExpectedError if the command has ended in an error.
+    """
+
+    def __init__(self, name: str) -> None:
+        self.timestamp = utils.current_tai()
+        self.name = name
+        self.ack_timestamp: float = 0.0
```

Comment:
Wouldn't it be better to have both `ack_timestamp` and `done_timestamp` as `None` to begin with? I know this makes typing harder but would make the interface cleaner. One thing you could do is have this as "private" attributes and make either a get method or an `attribute` to retrieve float values. 

Something like 

```
def get_ack_timestamp(self) -> float:
    if self._ack_timestamp is not None:
        return self._ack_timestamp
    else:
        raise RuntimeError(...)
```

Reply:
Actually, the timestamp isn't used yet so setting it to `None` is fine for now.